### PR TITLE
!join creates public rooms

### LINKED ISSUE #224: Joining channels via !join alters join rules
We've had a few cases where people are getting 403 Forbidden when trying to join `#freenode_` rooms. The `join_rules` state `invite` in this case, which is why they are getting 403d. They shouldn't ever be 403. I noticed this in one room today, and the person who joined the channel was doing so via `!join`. The `!join` command exists to make a user join a channel which is **invite-only** (since we nominally expect people to just join the room alias). I suspect the logic is a bit broken and is causing the room to be invite-only if the first(?) person to join it does so via `!join`.

### PR DESCRIPTION
**Pull request by lukebarnard1**
Should fix #224

### CODE REVIEW

**Comment by lukebarnard1** [comment]
And the tests are making sure the created rooms are in-fact private...

**Comment by Kegsay** [comment]
Yes because that is the desired behaviour when the rooms are not meant to be shared. We need to conditionally set them to public if the channel is allowed to be shared.

**Comment by Kegsay** [comment]
The code *should* be inspecting https://github.com/matrix-org/matrix-appservice-irc/blob/master/config.sample.yaml#L91 but it isn't.

**Comment by lukebarnard1** [comment]
That failed PR merge test seems to be caused by flaky tests... I'll try merging and seeing if I can make it a better merge and then push it here.

File: lib/bridge/MatrixHandler.js [hunk 1]
```diff
@@ -345,7 +345,7 @@ MatrixHandler.prototype._onAdminMessage = Promise.coroutine(function*(req, event
             ).createRoom({
                 options: {
                     name: ircChannel,
-                    visibility: "private"
+                    visibility: ircServer.getJoinRule()
```
Comment (Kegsay) [inline lib/bridge/MatrixHandler.js]: Join rule != visibility. Copy what we do on [alias creation](https://github.com/matrix-org/matrix-appservice-irc/blob/bdeebc0f81636037016adf6bd16224293d10cf66/lib/bridge/MatrixHandler.js#L1072):
```js
            let res = yield botIntent.createRoom({
                options: {
                    name: channelInfo.channel,
                    visibility: (
                        channelInfo.server.shouldPublishRooms() ? "public" : "private"
                    ),
                    creation_content: {
                        "m.federate": channelInfo.server.shouldFederate()
                    },
                    initial_state: [
                        {
                            type: "m.room.join_rules",
                            state_key: "",
                            content: {
                                join_rule: channelInfo.server.getJoinRule()
                            }
                        },
                        {
                            type: "m.room.history_visibility",
                            state_key: "",
                            content: {
                                history_visibility: "joined"
                            }
                        }
                    ]
                }
            });
```

Comment (lukebarnard1) [inline lib/bridge/MatrixHandler.js]: :+1:

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**